orchestrator/orchestrator/src/main_loop.rs

```diff
@@ -51,30 +51,37 @@ pub async fn orchestrator_main_loop(
         amount: 1u32.into(),
     };
 
-    let a = eth_oracle_main_loop(
+    let (tx, rx) = tokio::sync::mpsc::channel(1);
+
+    let a = send_main_loop(&contact, cosmos_key, fee.to_owned(), rx);
+
+    let b = eth_oracle_main_loop(
         cosmos_key,
         web3.clone(),
         contact.clone(),
         grpc_client.clone(),
         gravity_contract_address,
-        fee.clone(),
+        tx.clone(),
     );
-    let b = eth_signer_main_loop(
+
+    let c = eth_signer_main_loop(
         cosmos_key,
         ethereum_key,
         web3.clone(),
         contact.clone(),
         grpc_client.clone(),
         gravity_contract_address,
-        fee.clone(),
+        tx.clone(),
     );
-    let c = relayer_main_loop(
+
+    let d = relayer_main_loop(
         ethereum_key,
         web3,
         grpc_client.clone(),
         gravity_contract_address,
     );
-    join3(a, b, c).await;
+
+    futures::future::join4(a, b, c, d).await;
```

Comment:
Probably cleaner to switch to using the `join!` macro here.
https://rust-lang.github.io/async-book/06_multiple_futures/02_join.html

Reply:
Oh, that's nice, thanks for the pointer.

Reply:
Using this exposed a rust-analyzer issue: https://github.com/rust-analyzer/rust-analyzer/issues/6835

- cargo build --all && cargo test --all are both happy
- vs-code + rust-analyzer incorrectly think there's a problem:

```
{
	"resource": "/home/levi/PeggyJV/gravity-bridge/orchestrator/orchestrator/src/main_loop.rs",
	"owner": "rustc",
	"code": {
		"value": "unresolved-macro-call",
		"target": {
			"$mid": 1,
			"external": "https://rust-analyzer.github.io/manual.html#unresolved-macro-call",
			"path": "/manual.html",
			"scheme": "https",
			"authority": "rust-analyzer.github.io",
			"fragment": "unresolved-macro-call"
		}
	},
	"severity": 8,
	"message": "unresolved macro `$crate::join_internal!`",
	"source": "rust-analyzer",
	"startLineNumber": 85,
	"startColumn": 5,
	"endLineNumber": 85,
	"endColumn": 22
}
```

Probably best to stick with `join4` for now. Apparently the deep fix is:
https://github.com/rust-analyzer/rust-analyzer/issues/6835#issuecomment-845136858 (upgrade to futures >= 0.3.15

Reply:
For posterity, enabling this requires picking up the tokio macros feature:

```
-tokio = "1.4.0"
+tokio = { version = "1.4.0", features = ["macros"] }
```